File: kotlin/math/main_5086.kt
fun main() {
    while (true){
        val input = readLine()!!.split(' ').map { it.toInt() }
        if (input[0] == 0 && input[1] == 0)
            break
        if (input[0] > input[1] && input[0] % input[1] == 0)
            println("multiple")
        else if (input[0] < input[1] && input[1] % input[0] == 0)
            println("factor")
        else
            println("neither")
    }
}

File: cpp/math/main_5086.cpp
#include <iostream>
using namespace std;

int main() {
	int a, b;
	while (true) {
		cin >> a >> b;
		if (a == 0 && b == 0) {
			break;
		}
		if (a > b && a % b == 0) {
			cout << "multiple" << '\n';
		}
		else if (a < b && b % a == 0) {
			cout << "factor" << '\n';
		}
		else {
			cout << "neither" << '\n';
		}
	}
}